minio/crypto.py

```diff
@@ -20,16 +20,46 @@
 """Cryptography to read and write encrypted MinIO Admin payload"""
 
 import os
+import typing
 
 from argon2.low_level import Type, hash_secret_raw
 from Crypto.Cipher import AES, ChaCha20_Poly1305
 
-_NONCE_LEN = 8
+# Encrypted message format:
```

Comment:
Where do you get this information?

Reply:
From [your documentation](https://github.com/minio/madmin-go/blob/main/encrypt.go#L43-L49)

Reply:
Also it was present in the [current code](https://github.com/minio/minio-py/blob/master/minio/crypto.py#L27)

Reply:
And information that is present in the later part is present in the sio implementation:
* https://github.com/secure-io/sio-go/blob/master/sio.go#L25
* https://github.com/secure-io/sio/blob/master/img/channel_construction.svg
* https://github.com/secure-io/sio-go/tree/master

Reply:
1. Please point https://github.com/minio/madmin-go/blob/main/encrypt.go#L43-L49 instead of repeating it here.
2. I don't see more than 16k implementation in `madmin-go`. Could you point why do we need that?

Reply:
If you want to reproduce the issue try to create above 3k buckets with members (with long names). Then try to call an endpoint responsible for listing entities with their affiliation

Reply:
> If you want to reproduce the issue try to create above 3k buckets with members (with long names). Then try to call an endpoint responsible for listing entities with their affiliation

For this scenario, what API have you used to reproduce the issue?

Reply:
For example (`minioadmin`, so prefixed with proper path):
```
idp/ldap/policy-entities
idp/builtin/policy-entities
```

`list-users` for bigger instances…

Reply:
Anything with potentially more excessively capacious response